libuavcan_drivers/linux/include/uavcan_linux/socketcan.hpp

```diff
@@ -404,38 +433,46 @@ class SocketCanIface : public uavcan::ICanIface
             assert(0);
             return -1;
         }
-        std::vector< ::can_filter> filts(num_configs);
+        hw_filters_container_.clear();
+        hw_filters_container_.resize(num_configs);
+
         for (unsigned i = 0; i < num_configs; i++)
         {
             const uavcan::CanFilterConfig& fc = filter_configs[i];
-            filts[i].can_id   = fc.id   & uavcan::CanFrame::MaskExtID;
-            filts[i].can_mask = fc.mask & uavcan::CanFrame::MaskExtID;
+            hw_filters_container_[i].id   = fc.id   & uavcan::CanFrame::MaskExtID;
+            hw_filters_container_[i].mask = fc.mask & uavcan::CanFrame::MaskExtID;
             if (fc.id & uavcan::CanFrame::FlagEFF)
             {
-                filts[i].can_id |= CAN_EFF_FLAG;
+                hw_filters_container_[i].id |= CAN_EFF_FLAG;
             }
             if (fc.id & uavcan::CanFrame::FlagRTR)
             {
-                filts[i].can_id |= CAN_RTR_FLAG;
+                hw_filters_container_[i].id |= CAN_RTR_FLAG;
             }
             if (fc.mask & uavcan::CanFrame::FlagEFF)
             {
-                filts[i].can_mask |= CAN_EFF_FLAG;
+                hw_filters_container_[i].mask |= CAN_EFF_FLAG;
             }
             if (fc.mask & uavcan::CanFrame::FlagRTR)
             {
-                filts[i].can_mask |= CAN_RTR_FLAG;
+                hw_filters_container_[i].mask |= CAN_RTR_FLAG;
```

Comment:
You are mixing types from libuavcan and from SocketCAN, this is dangerous. The container `hw_filters_container_` contains objects of type `uavcan::CanFilterConfig`, you can't use SocketCAN constants with them; instead, you should use this: https://github.com/ilia-sheremet/uavcan/blob/master/libuavcan/include/uavcan/driver/can.hpp#L26-L30.